apis/devices/v1alpha2/zz_iothubfallbackroute_types.go

```diff
@@ -37,11 +37,25 @@ type IOTHubFallbackRouteParameters struct {
 	// +kubebuilder:validation:Required
 	Enabled *bool `json:"enabled" tf:"enabled,omitempty"`
 
-	// +kubebuilder:validation:Required
-	EndpointNames []*string `json:"endpointNames" tf:"endpoint_names,omitempty"`
+	// +crossplane:generate:reference:type=IOTHubEndpointStorageContainer
+	// +kubebuilder:validation:Optional
+	EndpointNames []*string `json:"endpointNames,omitempty" tf:"endpoint_names,omitempty"`
 
-	// +kubebuilder:validation:Required
-	IOTHubName *string `json:"iothubName" tf:"iothub_name,omitempty"`
+	// +kubebuilder:validation:Optional
+	EndpointNamesRefs []v1.Reference `json:"endpointNamesRefs,omitempty" tf:"-"`
+
+	// +kubebuilder:validation:Optional
+	EndpointNamesSelector *v1.Selector `json:"endpointNamesSelector,omitempty" tf:"-"`
+
+	// +crossplane:generate:reference:type=IOTHub
+	// +kubebuilder:validation:Optional
+	IOTHubName *string `json:"iothubName,omitempty" tf:"iothub_name,omitempty"`
```

Comment:
It's interesting that we have generated this field's serialized name as `iothubName` but the corresponding ref field's and selector field's serialized names as `iotHubNameRef` and `iotHubNameSelector`, respectively.